### DISCUSSION TITLE
Issue with external RAM?

### DISCUSSION THREAD

**Discussion by hsantens** [jomjol/AI-on-the-edge-device#1541] [General]
### The Problem

I'm new to this, and bought this dev board. https://www.amazon.de/gp/product/B0B3X6H9FN/ref=ppx_yo_dt_b_asin_title_o02_s00?ie=UTF8&th=1
After installing the firmware and adding the files/folders to the SD card, If get the following errors when checking the log (and the device does not seem to do anything)

E (893) psram: PSRAM ID read error: 0xffffffff
E (893) cpu_start: Failed to init external RAM!

Do you have any idea?

### Version

13.0.5

### Logfile

```shell
Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:7664
load:0x40078000,len:15360
load:0x40080400,len:3744
entry 0x4008067c
I (27) boot: ESP-IDF 4.3.2 2nd stage bootloader
I (27) boot: compile time 20:02:47
I (27) boot: chip revision: 3
I (30) boot_comm: chip revision: 3, min. bootloader chip revision: 0
I (37) boot.esp32: SPI Speed      : 40MHz
I (42) boot.esp32: SPI Mode       : DIO
I (46) boot.esp32: SPI Flash Size : 4MB
I (51) boot: Enabling RNG early entropy source...
I (56) boot: Partition Table:
I (60) boot: ## Label            Usage          Type ST Offset   Length
I (67) boot:  0 nvs              WiFi data        01 02 00009000 00004000
I (74) boot:  1 otadata          OTA data         01 00 0000d000 00002000
I (82) boot:  2 phy_init         RF data          01 01 0000f000 00001000
I (89) boot:  3 ota_0            OTA app          00 10 00010000 001db000
I (97) boot:  4 ota_1            OTA app          00 11 001f0000 001db000
I (104) boot: End of partition table
I (109) boot_comm: chip revision: 3, min. application chip revision: 0
I (116) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=4b88ch (309388) map
I (243) esp_image: segment 1: paddr=0005b8b4 vaddr=3ffb0000 size=04294h ( 17044) load
I (250) esp_image: segment 2: paddr=0005fb50 vaddr=40080000 size=004c8h (  1224) load
I (251) esp_image: segment 3: paddr=00060020 vaddr=400d0020 size=16668ch (1468044) map
I (819) esp_image: segment 4: paddr=001c66b4 vaddr=400804c8 size=1b1d0h (111056) load
I (867) esp_image: segment 5: paddr=001e188c vaddr=50000000 size=00010h (    16) load
I (881) boot: Loaded app from partition at offset 0x10000
I (881) boot: Disabling RNG early entropy source...
I (893) psram: This chip is ESP32-D0WD
E (893) psram: PSRAM ID read error: 0xffffffff
E (893) cpu_start: Failed to init external RAM!

abort() was called at PC 0x400829e1 on core 0


Backtrace:0x40083265:0x3ffe3ba00x4008f07d:0x3ffe3bc0 0x40096211:0x3ffe3be0 0x400829e1:0x3ffe3c50 0x4007956d:0x3ffe3c80  |<-CORRUPTED
```

**Comment by caco3** [comment]
Hi @hsantens 

Welcome!

Please search the discussions for this: https://github.com/jomjol/AI-on-the-edge-device/discussions?discussions_q=PSRAM%20ID%20read%20error

It looks like your hardware is broken!

**Reply by hsantens** [reply]
Thanks, so this "Built-in 520KB SRAM, External 8MB PSRAM" should not be a problem? It should work (if hardware isn't broken)?

**Reply by caco3** [reply]
It should, but with those Chinese devices you never know if the description is correct. This is an issue for many devices at the moment, sadly (see other discussions).

**Comment by bucklevision** [comment]
I am having the same issue. I bought 3 boards, 2 from one supplier and one from the other. All look good quality, well soldered. All 3 boards accept programming through the web interface or the esptool route and show no errors, however only one (one of the pair) actually accepted the firmware, the other two both return the same error you see. All 3 were sold as having PSRAM, with the same board spec. Tonight I will try swapping the CH340 "motherboard" because the two that fail both stop streaming logs to me and their port vanishes after 30 seconds or so, whereas the working one doesn't. That's my only clue so far.
